src/gc-alloc-profiler.cpp

```diff
@@ -20,6 +20,8 @@ struct jl_raw_alloc_t {
     jl_datatype_t *type_address;
     jl_raw_backtrace_t backtrace;
     size_t size;
+    jl_task_t *task;
+    uint64_t timestamp;
```

Comment:
It's nit-picking, but I would appreciate if you either stored the task as a `void*`, or even a `size_t`, and/or added a comment here indicating that this pointer is not rooted, and therefore might not even be live anymore, and you should never try to load it.

Can you send a tiny follow up to address this? :) Thanks!

I also like matching the julia and C structs as much as possible, so void* in both or size_t in both makes sense to me.